### PR TITLE
make retry self-contained and add pkg to get EC2 user-data

### PR DESCRIPTION
**Pull request by 4z3**

### CODE REVIEW

File: pkgs/retry/retry [hunk 1]
```diff
@@ -4,6 +4,8 @@
 #
 set -euf
 
+export PATH=@coreutils@/bin"${PATH+:$PATH}"
```
Comment (phunehehe) [inline pkgs/retry/retry:7]: I haven't seen this markup before (`@abc@`). What is it called?

Comment (4z3) [inline pkgs/retry/retry:7]: I don't know its name, but it's used by [substituteAll](https://github.com/zalora/microgram/pull/16/files#diff-9d168b32227c60d3951d0fd8cf125d22R11).

Comment (ip1981) [inline pkgs/retry/retry:7]: http://www.gnu.org/software/autoconf/manual/autoconf-2.61/html_node/Makefile-Substitutions.html :-)

Comment (phunehehe) [inline pkgs/retry/retry:7]: Cool Igor, but that prompts the next question: what sets `coreutils` here? Some `configure`?

Comment (4z3) [inline pkgs/retry/retry:7]: Beside the syntax, it has nothing to do with Makefile. Have a look at [the documentation of substituteAll](https://nixos.org/nixpkgs/manual/#fun-substituteAll) and, to see how every attribute (e.g. [coreutils](https://github.com/zalora/microgram/pull/16/files#diff-9d168b32227c60d3951d0fd8cf125d22R7)) is passed as an environment variable to the builder, see [NixOS's derivation section](http://nixos.org/nix/manual/#ssec-derivation).

**Comment by phunehehe** [comment]
OK, thanks for the links :)

**Pull request merged**